packages/flutter/lib/src/material/ink_sparkle.dart

```diff
@@ -0,0 +1,3667 @@
+/// the well rectangle, then clipped to it when drawn. The well
+/// rectangle is the box returned by [rectCallback], if provided, or
+/// otherwise is the bounds of the [referenceBox].
+///
+/// If [containedInkWell] is false, then [rectCallback] should be null.
+/// The ink ripple is clipped only to the edges of the [Material].
+/// This is the default.
+///
+/// When the ripple is removed, [onRemoved] will be called.
+///
+/// {@tool snippet}
+///
+/// For typical use, pass the [InkSparkle.splashFactory] to the `splashFactory`
+/// parameter of a button style or [ThemeData].
+///
+/// ```dart
+/// ElevatedButton(
+///   style: ElevatedButton.styleFrom(splashFactory: InkSparkle.splashFactory),
+///   child: const Text('Sparkle!'),
+///   onPressed: () { },
+/// )
+/// ```
+/// {@end-tool}
+class InkSparkle extends InteractiveInkFeature {
+  /// Begin a sparkly ripple effect, centered at [position] relative to
+  /// [referenceBox].
+  ///
+  /// The [color] defines the color of the splash itself. The sparkles are
+  /// always white.
+  ///
+  /// The [controller] argument is typically obtained via
+  /// `Material.of(context)`.
+  ///
+  /// [textDirection] is used by [customBorder] if it is non-null. This allows
+  /// the [customBorder]'s path to be properly defined if it was the path was
+  /// expressed in terms of "start" and "end" instead of
+  /// "left" and "right".
+  ///
+  /// If [containedInkWell] is true, then the ripple will be sized to fit
+  /// the well rectangle, then clipped to it when drawn. The well
+  /// rectangle is the box returned by [rectCallback], if provided, or
+  /// otherwise is the bounds of the [referenceBox].
+  ///
+  /// If [containedInkWell] is false, then [rectCallback] should be null.
+  /// The ink ripple is clipped only to the edges of the [Material].
+  /// This is the default.
+  ///
+  /// Clipping can happen in 3 different ways:
+  ///  1. If [customBorder] is provided, it is used to determine the path for
+  ///     clipping.
+  ///  2. If [customBorder] is null, and [borderRadius] is provided, the canvas
+  ///     is clipped by an [RRect] created from `_clipCallback` and
+  ///     [borderRadius].
+  ///  3. If [borderRadius] is the default [BorderRadius.zero], then the [Rect]
+  ///     provided by `_clipCallback` is used for clipping.  /// Clipping can happen in 3 different ways:
+  ///  1. If [customBorder] is provided, it is used to determine the path for
+  ///     clipping.
+  ///  2. If [customBorder] is null, and [borderRadius] is provided, the canvas
+  ///     is clipped by an [RRect] created from `_clipCallback` and
+  ///     [borderRadius].
+  ///  3. If [borderRadius] is the default [BorderRadius.zero], then the [Rect]
+  ///     provided by `_clipCallback` is used for clipping.
+  ///
+  /// When the ripple is removed, [onRemoved] will be called.
+  ///
+  /// [turbulenceSeed] can be passed if a non random seed shold be used for
```

Comment:
nit: docs should be grammatically correct and therefore sentences should start with a capital letter. May have to restructure this sentence a little bit to not have it start with the argument name.